src/App.tsx

```diff
@@ -25,26 +25,23 @@ export const App: FC = () => {
   const [error, setError] = useState(false);
   const [isLoading, setIsLoading] = useState(false);
 
-  const visibleTodos = useMemo(() => {
-    return getFilteredTodos(todos, filterType, query);
-  }, [todos, filterType, query]);
+  const visibleTodos = getFilteredTodos(todos, filterType, query);
 
-  useEffect(() => {
-    const fetchData = async () => {
-      setIsLoading(true);
+  const fetchData = async () => {
+    setIsLoading(true);
```

Comment:
```suggestion
    setError(false);
    setIsLoading(true);
```
Reset error. If you, for example, add reload button it will be useful.